spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -15,34 +15,39 @@
  */
 package org.apache.spark.ml
 
-import scala.reflect.ClassTag
-import com.intel.analytics.bigdl.dataset.{DataSet, MiniBatch}
 import com.intel.analytics.bigdl.{Criterion, Module}
-import com.intel.analytics.bigdl.optim.{Adam, Optimizer, Trigger}
-import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.dataset.{DataSet, MiniBatch}
+import com.intel.analytics.bigdl.models.utils.ModelBroadcast
+import com.intel.analytics.bigdl.optim._
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import org.apache.spark.ml.param.{IntParam, ParamMap, ParamValidators}
-import org.apache.spark.sql.DataFrame
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import org.apache.spark.ml.param.{IntParam, ParamMap, ParamValidators, _}
+import org.apache.spark.ml.util.SchemaUtils
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Row}
+
+import scala.reflect.ClassTag
 
 /**
  * [[DLEstimator]] helps to train a BigDL Model with the Spark ML Estimator/Transfomer pattern,
  * thus Spark users can conveniently fit BigDL into Spark ML pipeline.
  *
- * The feature column holds the storage (Spark Vectors or array of Floats or Doubles) of
- * the feature data, and user should specify the tensor size(dimensions) via param featureSize.
- * The label column holds the storage (Spark Vectors, array of Floats or Doubles, or Double) of
- * the label data, and user should specify the tensor size(dimensions) via param labelSize.
- * Internally the feature and label data are converted to BigDL tensors, to further train a
- * BigDL model efficiently.
+ * [[DLEstimator]] supports feature and label data in the format of Array[Double], Array[Float],
+ * org.apache.spark.mllib.linalg.{Vector, VectorUDT} for Spark 1.5, 1.6 and
+ * org.apache.spark.ml.linalg.{Vector, VectorUDT} for Spark 2.0+. Also label data can be of
+ * DoubleType.
+ * User should specify the feature data dimensions and label data dimensions via the constructor
+ * parameters featureSize and labelSize respectively. Internally the feature and label data are
+ * converted to BigDL tensors, to further train a BigDL model efficiently.
  *
- * For details usage, please refer to example :
- * [[com.intel.analytics.bigdl.example.MLPipeline.DLEstimatorLeNet]]
+ * For details usage, please refer to examples in package
+ * com.intel.analytics.bigdl.example.MLPipeline
```

Comment:
I'm afraid the package name may break the javadoc generation if some day we want to release the API doc.